examples/consistency_models/train_consistency_distillation.py

```diff
@@ -0,0 +1,724 @@
+    logger.info(f"  Total optimization steps = {max_train_steps}")
+
+    global_step = 0
+    first_epoch = 0
+
+    # Potentially load in the weights and states from a previous save
+    if args.resume_from_checkpoint:
+        if args.resume_from_checkpoint != "latest":
+            path = os.path.basename(args.resume_from_checkpoint)
+        else:
+            # Get the most recent checkpoint
+            dirs = os.listdir(args.output_dir)
+            dirs = [d for d in dirs if d.startswith("checkpoint")]
+            dirs = sorted(dirs, key=lambda x: int(x.split("-")[1]))
+            path = dirs[-1] if len(dirs) > 0 else None
+
+        if path is None:
+            accelerator.print(
+                f"Checkpoint '{args.resume_from_checkpoint}' does not exist. Starting a new training run."
+            )
+            args.resume_from_checkpoint = None
+        else:
+            accelerator.print(f"Resuming from checkpoint {path}")
+            accelerator.load_state(os.path.join(args.output_dir, path))
+            global_step = int(path.split("-")[1])
+
+            initial_global_step = global_step * args.gradient_accumulation_steps
+            first_epoch = global_step // num_update_steps_per_epoch
+    else:
+        initial_global_step = 0
+
+    progress_bar = tqdm(
+        range(0, max_train_steps),
+        initial=initial_global_step,
+        desc="Steps",
+        # Only show the progress bar once on each machine.
+        disable=not accelerator.is_local_main_process,
+    )
+
+    timesteps = noise_scheduler.timesteps 
+    sigmas = noise_scheduler.sigmas # in reverse order, sigma0 is sigma_max
+
+
+    # Train!
+    for epoch in range(first_epoch, args.num_epochs):
+        model.train()
+        for step, batch in enumerate(train_dataloader):
+            clean_images = batch["input"]
+            labels = batch["labels"]
+            # Sample noise that we'll add to the images
+            noise = torch.randn(clean_images.shape).to(clean_images.device)
+            # Sample a random timestep for each image, TODO - allow different timesteps in a batch
+            index = torch.randint(
+                0, noise_scheduler.config.num_train_timesteps-1,  (1,), device=clean_images.device
+            ).long()
+            # timestep is the scaled timestep, sigma is the unscaled timestep
+            timestep = timesteps[index+1]
+            sigma = sigmas[index+1]
+            timestep_prev = timesteps[index]
+            sigma_prev = sigmas[index]
+            # add noise expects the scaled timestep only and internally converts to sigma
+            noised_image = noise_scheduler.add_noise(clean_images, noise, timestep)
+            target_model_ema.copy_to(target_model.parameters())
+
+            with accelerator.accumulate(model):
+                # Predict the noise residual
+                model_output = model(noise_scheduler.scale_model_input(noised_image, timestep), timestep, class_labels=labels).sample
```

Comment:
There are three `model_output` variables here. Let's maybe try to give them more appropriate names.

Reply:
See https://github.com/huggingface/diffusers/pull/3992#discussion_r1259066402 for some suggestions.